apps/ecommerce/frontend/src/components/Field/ResourceCard/useResourceMenuItems.tsx

```diff
@@ -0,0 +1,57 @@
+import { Menu } from '@contentful/f36-components';
+import { useEffect, useMemo, useState } from 'react';
+import type { ResourceCardMenuProps } from './ResourceCardMenu';
+
+type menuItemsOutput = { menuItems: JSX.Element[] };
+
+const useResourceMenuItems = (menuActions: ResourceCardMenuProps): menuItemsOutput => {
+  const defaultMenuItems = useMemo(
+    () => [
+      <Menu.Item key="delete" onClick={() => menuActions.onRemove()}>
+        Remove
+      </Menu.Item>,
+    ],
+    [menuActions]
+  );
+
+  const [menuItems, setMenuItems] = useState([...defaultMenuItems]);
+
+  const addMenuItems = () => {
+    const newMenuItems = [...defaultMenuItems];
+
+    const { index, total, onMoveToTop, onMoveToBottom } = menuActions;
+    if (typeof index !== 'undefined' && total && total > 1) {
+      if (index > 0 && onMoveToTop) {
+        newMenuItems.push(
+          <Menu.Item key="moveToTop" onClick={() => onMoveToTop()}>
+            Move to top
+          </Menu.Item>
+        );
+      }
+
+      if (index + 1 !== total && onMoveToBottom) {
```

Comment:
Can't you just do ```index + 1 < total```?

Also why do we even keep a total track? Isn't the total just the menuItems array length?